runtime/lua/vim/lsp.lua

```diff
@@ -965,6 +965,7 @@ function lsp.start_client(config)
     changetracking.reset(client_id)
     if code ~= 0 or (signal ~= 0 and signal ~= 15) then
       local msg = string.format('Client %s quit with exit code %s and signal %s', client_id, code, signal)
+      local _ = log.warn() and log.warn(log_prefix, msg)
```

Comment:
certainly should log such things. Is this dummy assignment a common pattern? why can't we just call `log.warn(log_prefix, msg)`

Reply:
Agreed it looks really weird. I was following the conventions I found elsewhere. When calling a log function with no args, it returns `true` if  the current log level permits logging. This is however already guarded by [this branch though](https://github.com/williamboman/neovim/blob/3987466c0590ab3664d8ee917eb5ef7c621d9407/runtime/lua/vim/lsp/log.lua#L91-L93), so I'm not entirely sure what the rationale is behind doing this.

Reply:
Nvm, I just read the docstring 5 lines above what I linked, the rationale is:

> -- This way you can avoid string allocations if the log level isn't high enough.